### ISSUE TITLE
Crash If Main Window Closed (MainWindow::updateTouchBar())

### ISSUE THREAD

**Issue by philipbel** [gitahead/gitahead#370]
GitAhead crashes in MainWindow::updateTouchBar().

- Mac: 2015 Retina MacBook Pro (no touchbar)
- macOS Catalina 10.15.3 (19D76)
- GitAhead version v2.6.1.

Occurs always if GitAhead is closed by closing the main window. If closed using Quit (or Cmd-Q), no crash is observed. If launched from the command line, the following is printed:

```
% ga
/usr/local/bin/ga: line 2: 29884 Segmentation fault: 11  /Applications/GitAhead.app/Contents/MacOS/GitAhead $@
```

If launched from the Dock or the Finder, the macOS Crash system dialog is shown, providing a stacktrace (see below).

```
[...]
Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000018
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [30208]
[...]
Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   libobjc.A.dylib               	0x00007fff64df001d objc_msgSend + 29
1   com.gitahead.GitAhead         	0x000000010eb5d6c9 MainWindow::updateTouchBar(int, int) + 57
2   com.gitahead.GitAhead         	0x000000010eb1551d MainWindow::updateInterface() + 365
3   com.gitahead.GitAhead         	0x000000010eb182a9 QtPrivate::QFunctorSlotObject<MainWindow::MainWindow(git::Repository const&, QWidget*, QFlags<Qt::WindowType>)::$_3, 1, QtPrivate::List<int>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*) + 41
4   org.qt-project.QtCore         	0x000000011015753b QMetaObject::activate(QObject*, int, int, void**) + 2219
5   org.qt-project.QtWidgets      	0x000000010f6068ad 0x10f427000 + 1964205
6   org.qt-project.QtCore         	0x00000001101578cc QMetaObject::activate(QObject*, int, int, void**) + 3132
7   org.qt-project.QtWidgets      	0x000000010f5e8e93 QTabBar::removeTab(int) + 1123
8   org.qt-project.QtWidgets      	0x000000010f6068ee 0x10f427000 + 1964270
9   org.qt-project.QtCore         	0x00000001101578cc QMetaObject::activate(QObject*, int, int, void**) + 3132
10  org.qt-project.QtWidgets      	0x000000010f5e306e 0x10f427000 + 1818734
11  org.qt-project.QtCore         	0x00000001101578cc QMetaObject::activate(QObject*, int, int, void**) + 3132
12  org.qt-project.QtWidgets      	0x000000010f45aeb6 QStackedLayout::takeAt(int) + 438
13  org.qt-project.QtWidgets      	0x000000010f4545cb 0x10f427000 + 185803
14  org.qt-project.QtWidgets      	0x000000010f45412e QLayout::widgetEvent(QEvent*) + 126
15  org.qt-project.QtWidgets      	0x000000010f43775d QApplicationPrivate::notify_helper(QObject*, QEvent*) + 237
16  org.qt-project.QtWidgets      	0x000000010f438b82 QApplication::notify(QObject*, QEvent*) + 594
17  org.qt-project.QtCore         	0x0000000110126574 QCoreApplication::notifyInternal2(QObject*, QEvent*) + 212
18  org.qt-project.QtCore         	0x000000011014f833 QObjectPrivate::setParent_helper(QObject*) + 387
19  org.qt-project.QtCore         	0x000000011014f416 QObject::~QObject() + 1734
20  org.qt-project.QtWidgets      	0x000000010f4624d4 QWidget::~QWidget() + 1188
21  com.gitahead.GitAhead         	0x000000010eb2ecee RepoView::~RepoView() + 14
22  org.qt-project.QtCore         	0x000000011014fd22 QObject::event(QEvent*) + 146
23  org.qt-project.QtWidgets      	0x000000010f47457a QWidget::event(QEvent*) + 4746
24  org.qt-project.QtWidgets      	0x000000010f43777d QApplicationPrivate::notify_helper(QObject*, QEvent*) + 269
25  org.qt-project.QtWidgets      	0x000000010f438b82 QApplication::notify(QObject*, QEvent*) + 594
26  org.qt-project.QtCore         	0x0000000110126574 QCoreApplication::notifyInternal2(QObject*, QEvent*) + 212
27  org.qt-project.QtCore         	0x00000001101277ae QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) + 878
28  libqcocoa.dylib               	0x00000001121a0b29 0x11216b000 + 219945
29  libqcocoa.dylib               	0x00000001121a1298 0x11216b000 + 221848
30  com.apple.CoreFoundation      	0x00007fff2eac6b21 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
31  com.apple.CoreFoundation      	0x00007fff2eac6ac0 __CFRunLoopDoSource0 + 103
32  com.apple.CoreFoundation      	0x00007fff2eac68d4 __CFRunLoopDoSources0 + 209
33  com.apple.CoreFoundation      	0x00007fff2eac5740 __CFRunLoopRun + 1272
34  com.apple.CoreFoundation      	0x00007fff2eac4bd3 CFRunLoopRunSpecific + 499
35  com.apple.HIToolbox           	0x00007fff2d61a65d RunCurrentEventLoopInMode + 292
36  com.apple.HIToolbox           	0x00007fff2d61a2a9 ReceiveNextEventCommon + 356
37  com.apple.HIToolbox           	0x00007fff2d61a127 _BlockUntilNextEventMatchingListInModeWithFilter + 64
38  com.apple.AppKit              	0x00007fff2bc8aba4 _DPSNextEvent + 990
39  com.apple.AppKit              	0x00007fff2bc89380 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1352
40  com.apple.AppKit              	0x00007fff2bc7b09e -[NSApplication run] + 658
41  libqcocoa.dylib               	0x00000001121a0013 0x11216b000 + 217107
42  org.qt-project.QtCore         	0x0000000110121b8f QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) + 431
43  org.qt-project.QtCore         	0x0000000110126b82 QCoreApplication::exec() + 130
44  com.gitahead.GitAhead         	0x000000010eab409a main + 154
45  libdyld.dylib                 	0x00007fff6616a7fd start + 1
[...]
```

**Comment by hackhaslam** [comment]
Fixed in 56b1f47375db72f2be6ccaafd73e9acc7d0aec27.

**Comment by hackhaslam** [comment]
This should now be fixed in 2.6.2.
